src/storage/datasets_index.rs

```diff
@@ -43,6 +44,60 @@ impl DatasetsIndex {
                 .collect()
         })
     }
+    pub fn from(assigned_data: Vec<crate::util::assignment::Dataset>, headers: HashMap<String, String>) -> Self {
+        let mut datasets = HashMap::new();
+        let mut chunks_ordinals_map = HashMap::new();
+        let mut ordinal = 0;
+        for dataset in assigned_data {
+            let dataset_id = dataset.id;
+            let dataset_url = dataset.base_url;
+            let mut dataset_files: HashMap<DataChunk, Vec<Arc<str>>> = Default::default();
+            for chunk in dataset.chunks {
+                let data_chunk = DataChunk::from_path(&chunk.id).unwrap();
+                let mut files: Vec<Arc<str>> = Default::default();
+                for (file, _) in chunk.files {
+                    files.push(Arc::from(file.as_str()));
```

Comment:
This way there is no point in wrapping the filenames in `Arc`s. Let's either remove Arcs and explicitly store `String`s or reuse the filenames if they were already encountered

Reply:
done